tests/test_algorithm/test_id_alg.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+
+"""Tests for the identify algorithm."""
+
+import unittest
+
+from y0.algorithm.identify import identify
+from y0.algorithm.taheri_design import _get_result, iterate_lvdags
+from y0.dsl import Expression, P, Sum, X, Y, Z, W1, W2, Y1, Y2
+from y0.graph import DEFAULT_TAG, NxMixedGraph, admg_to_latent_variable_dag
+from y0.mutate import canonicalize
+from y0.parser import parse_craig
+from y0.resources import VIRAL_PATHOGENESIS_PATH
+
+P_XY = P(X, Y)
+P_XYZ = P(X, Y, Z)
+
+
+class TestIdentify(unittest.TestCase):
+    """Test cases from https://github.com/COVID-19-Causal-Reasoning/Y0/blob/master/ID_whittemore.ipynb."""
+
+    def assert_expr_equal(self, expected: Expression, actual: Expression):
+        """Assert that two expressions are the same"""
+        expected_vars = expected.get_variables()
+        self.assertEqual(expected_vars, actual.get_variables())
+        ordering = list(expected_vars)
+        expected_canonical = canonicalize(expected, ordering)
+        actual_canonical = canonicalize(actual, ordering)
+        self.assertEqual(
+            expected_canonical,
+            actual_canonical,
+            msg=f"\nExpected: {str(expected_canonical)}\nActual:   {str(actual_canonical)}",
+        )
+
+    def assert_identify(
+        self, expression: Expression, graph: NxMixedGraph, query: Expression
+    ):
+        """Assert that the graph returns the same."""
+        self.assert_expr_equal(expression, identify(graph, query))
+
+    def test_figure_2a(self):
+        """Test Figure 2A."""
```

Comment:
Would you please add the reference from which these figures' graphs are coming in the docstrings

Reply:
Done

Reply:
Shpitser, I., & Pearl, J. (2008). Complete Identification Methods for the Causal Hierarchy. Journal of Machine Learning Research.